kubernetes/heapster
# Updates to the Hawkular-Metrics sink

This PR adds some changes to the Hawkular-Metrics sink described below. Some of the changes are done to provide better support for using Heapster on the Openshift v3, while others are fixes that did not work correctly in the first version. Do not merge yet, requires still an update to the Hawkular-Metrics client (to released version) - but open to comments.

New features:

* Add support for Cumulative metric type
* Add feature to transform metric label value to Hawkular-Metrics tenant

Tests:

* Add integration tests to check Register and Timeseries methods.

Improvements:

* Parallel Storetimeseries operations to improve performance when storing to multiple tenants
* Make registerIfNecessary happen in the background goroutine

Fixes:

* Fix registerIfNecessary leaking map reference between metrics. 

Documentation:

* Improve documentation of Hawkular-Metrics sink and add link to instructions on deploying Heapster + Hawkular-Metrics combination to Openshift (kubernetes.json should work on plain Kubernetes also with small changes)
